platform/pulpcore/app/models/content.py

```diff
@@ -38,36 +76,47 @@ def natural_key(self):
         return tuple(getattr(self, f.name) for f in self.natural_key_fields)
 
 
-class Artifact(Model):
+class ContentArtifact(Model):
     """
-    A file associated with a piece of content.
+    A relationship between a Content and an Artifact.
 
-    Fields:
+    Serves as a through model for the 'artifacts' ManyToManyField in Content.
+    """
+    artifact = models.ForeignKey(Artifact, on_delete=models.CASCADE, null=True)
+    content = models.ForeignKey(Content, on_delete=models.CASCADE)
+    relative_path = models.CharField(max_length=64)
 
-        file (models.FileField): The stored file.
-        size (models.IntegerField): The size of the file in bytes.
-        md5 (models.CharField): The MD5 checksum of the file.
-        sha1 (models.CharField): The SHA-1 checksum of the file.
-        sha224 (models.CharField): The SHA-224 checksum of the file.
-        sha256 (models.CharField): The SHA-256 checksum of the file.
-        sha384 (models.CharField): The SHA-384 checksum of the file.
-        sha512 (models.CharField): The SHA-512 checksum of the file.
+    class Meta:
+        unique_together = ('artifact', 'content', 'relative_path')
```

Comment:
@dkliban , as per our discussion, let's not add `artifact` to the unique index, `content` and `relative_path` should be enough.
Because `artifact` can be `null` and is a part of this unique index, we can end up with with the duplicated (`relative_path`, `content`) in case of the sync of the same content with different download policies.

Example:
- Create repo with on_demand policy. Sync.
- Create another repo with the same feed and immediate policy. Sync.
- Check DB: https://paste.fedoraproject.org/paste/rbkT1gjqKzNGBgcRvzgLQQ